apis/externalsecrets/v1beta1/secretstore_beyondtrust_types.go

```diff
@@ -0,0 +1,63 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import esmeta "github.com/external-secrets/external-secrets/apis/meta/v1"
+
+type BeyondTrustProviderSecretRef struct {
+
+	// Value can be specified directly to set a value without using a secret.
+	// +optional
+	Value string `json:"value,omitempty"`
+
+	// SecretRef references a key in a secret that will be used as value.
+	// +optional
+	SecretRef *esmeta.SecretKeySelector `json:"secretRef,omitempty"`
+}
+
+// Configures a store to sync secrets using BeyondTrust Password Safe.
+type BeyondtrustAuth struct {
+	// +required - API OAuth Client ID.
+	ClientID *BeyondTrustProviderSecretRef `json:"clientid"`
+	// +required - API OAuth Client Secret.
+	ClientSecret *BeyondTrustProviderSecretRef `json:"clientsecret"`
+	// Content of the certificate (cert.pem) for use when authenticating with an OAuth client Id using a Client Certificate.
+	Certificate *BeyondTrustProviderSecretRef `json:"certificate,omitempty"`
+	// Certificate private key (key.pem). For use when authenticating with an OAuth client Id
+	CertificateKey *BeyondTrustProviderSecretRef `json:"certificatekey,omitempty"`
+}
+
+// Configures a store to sync secrets using BeyondTrust Password Safe.
+type BeyondtrustServer struct {
+	// +required - BeyondTrust Password Safe API URL. https://example.com:443/beyondtrust/api/public/V3.
+	APIURL string `json:"apiurl"`
+	// The secret retrieval type. SECRET = Secrets Safe (credential, text, file). MANAGED_ACCOUNT = Password Safe account associated with a system.
+	RetrievalType string `json:"retrievaltype,omitempty"`
```

Comment:
You haven't change these to camelCase. Please use `retrievalType` `verifyCA` `clientTimeoutSeconds` etc..

Reply:
You mean, it should be likec?

	RetrievalType string `json:"**retrievalType**,omitempty"`
	VerifyCA bool `json:"**verifyCA**"`
	ClientTimeOutSeconds int `json:"**clientTimeOutSseconds**,omitempty"`

Reply:
>Should I change json variables as well ?

Correct. Those will be keys in the yaml later on and the rest of the providers use a camelCase pattern for these. :)

Reply:
Hi @Skarlso solved, please check when you have then chance, thank you!